commit ab02684302c42414f2920cb45db338c8a347208f (refs/changes/06/606/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-10-05T19:28:46-07:00 (3 years ago)
    
    joyent/node-sshpk#17 want tests covering PKCS8 private key format

diff --git a/test/assets/id_dsa8 b/test/assets/id_dsa8
new file mode 100644
index 0000000..7329838
--- /dev/null
+++ b/test/assets/id_dsa8
@@ -0,0 +1,9 @@
+-----BEGIN PRIVATE KEY-----
+MIIBSwIBADCCASwGByqGSM44BAEwggEfAoGBANBGJdbHdT5d/1tU3d3Cw1Hr2/Po
+yYsmaGP82wH+2apK0RoZOmQfHlc/oyz9yordFB/VgleR800m5rJJH2TvwMqphw61
+1oGDu+hSqV5UGyQBKIzPfh4mi9WHY4rZ6nHEeC2SiT6QKZG7dutDvXHmjBrLU+hw
+BY/Fd7oBN9P0Qxu3AhUAtP5o3ksvBJ8DLlITtPfEGigW2BUCgYEAtZiJz9yGBwT1
+vsoLM938qkWSYU78qseOqXmpXFgCi9nbtXWvrLFEMAsQIUTBzNkEyDWyVBU8ld9y
+cvlaj5jclNaRJ/JLcGyzK/8I78/QvtiY/VkSxLxymOkLI33q/yXWn5yUEU5HyBdi
+yI4eLFvYS2slyxYWq63b6NTr99lFBd4EFgIUeSmuyEZnBHa/StOvH9o2xE8MztE=
+-----END PRIVATE KEY-----
diff --git a/test/assets/id_ecdsa8 b/test/assets/id_ecdsa8
new file mode 100644
index 0000000..fed189f
--- /dev/null
+++ b/test/assets/id_ecdsa8
@@ -0,0 +1,6 @@
+-----BEGIN PRIVATE KEY-----
+MIG2AgEAMBAGByqGSM49AgEGBSuBBAAiBIGeMIGbAgEBBDCNOaMOKbtjXYQykwv+
+FCUUi2jd5ojUv4QZ3BhoBPEse3uEaSlWrYSyNnxiulPa3TehZANiAAQIVUh8EgyM
+x3mJtkVSYQiH8eG+sK8Wvt8s6y9r3XaHgnRNoZ4ccvY64dcVgdFFOhn1tlIj7mQY
+h1ptmIIxd8jtpNi/uBPy4w3NlvD2oCnA/3VnEnBM6QaKS+EK92HMyYM=
+-----END PRIVATE KEY-----
diff --git a/test/assets/id_rsa8 b/test/assets/id_rsa8
new file mode 100644
index 0000000..6649441
--- /dev/null
+++ b/test/assets/id_rsa8
@@ -0,0 +1,16 @@
+-----BEGIN PRIVATE KEY-----
+MIICdwIBADANBgkqhkiG9w0BAQEFAASCAmEwggJdAgEAAoGBANWaXdgNWqKYXJWA
+ryuAQSK0D74arruteq8pXjRXkaVPs4ZAv6LTOtfyKQtHwikK9VyzfsXvwqiEX3AO
+PznieRxQJs2HQmKGivcjlpiWm8QWU7hLyeNXD0o6tj4v3/QAtEArpQ73qBPBE5bm
+j/yC3IPFQnEQXYOJbN6C6ZAlbVI3AgMBAAECgYAFNv+Thr/vuEiKXdno9fihIERC
+H+SkMFgObrQeEwU7Q/beEDadgwqsuvzFibrzxZC38qnlqzTVaGXsFmz0vNK4EtCy
+cN8L6T2u/uU8rgtrdthf9PvLVXoTKmjwhORQmCkZE3BQzMsZL1NWkuz6CeUm6bt5
+K+QY/9k4Np1ElTcSAQJBAPB6RlBKYI6HZv3JfxebMRXfNWi1jHwgAKqGK2g25zy2
+IZu8uodk/MnceC7pFj3nGxqh+ezJA0dE/vxaN0n6vVsCQQDjZAHYAsMz6zSI9sR9
+tNAUUiSW9ZGIJjl8VLmG4w1UXKQ8fCGzenQEklYJAPw1umZI9wpw0SiWR9z+AMFr
+gMlVAkEAqCp53VAF8eYSScuMyu9GBiuPZ+f9lBe1H4qv9DLgxf3/xq7WKjSe6nL4
+wKVTXnqs45T9924iksZXIhXcsCleHQJAT2K6wg801wtma+q67qgRfQawf4N8K2ws
+5ZsuqkjafnlZXUR4/4/gsTflkX8mjhtlWlPk8T4UTWMJZCQfYsizgQJBAI7Ul+68
+kjI/8g6+N0LWYd090WHhQt0YkNvcp6CTdqek9prSO7Wp9BzV1Y5ql590v7+jdka4
+V/nl+Goh1LQ55iQ=
+-----END PRIVATE KEY-----
diff --git a/test/private-key.js b/test/private-key.js
index a5a0275..8c274da 100644
--- a/test/private-key.js
+++ b/test/private-key.js
@@ -114,6 +114,34 @@ test('PrivateKey convert ssh-private rsa to pem', function (t) {
 	t.end();
 });
 
+test('parse pkcs8 unencrypted private keys', function (t) {
+	var keyPkcs8 = fs.readFileSync(path.join(testDir, 'id_rsa8'));
+	var key = sshpk.parsePrivateKey(keyPkcs8, 'pkcs8');
+	t.strictEqual(key.type, 'rsa');
+	t.ok(ID_RSA_FP.matches(key));
+
+	var newPkcs8 = key.toBuffer('pkcs8');
+	t.strictEqual(keyPkcs8.toString(), newPkcs8.toString());
+
+	keyPkcs8 = fs.readFileSync(path.join(testDir, 'id_ecdsa8'));
+	key = sshpk.parsePrivateKey(keyPkcs8, 'pkcs8');
+	t.strictEqual(key.type, 'ecdsa');
+	t.ok(ID_ECDSA_FP.matches(key));
+
+	newPkcs8 = key.toBuffer('pkcs8');
+	t.strictEqual(keyPkcs8.toString(), newPkcs8.toString());
+
+	keyPkcs8 = fs.readFileSync(path.join(testDir, 'id_dsa8'));
+	key = sshpk.parsePrivateKey(keyPkcs8, 'pkcs8');
+	t.strictEqual(key.type, 'dsa');
+	t.ok(ID_DSA_FP.matches(key));
+
+	newPkcs8 = key.toBuffer('pkcs8');
+	t.strictEqual(keyPkcs8.toString(), newPkcs8.toString());
+
+	t.end();
+});
+
 test('parse and produce encrypted ssh-private ecdsa', function (t) {
 	var keySsh = fs.readFileSync(path.join(testDir, 'id_ecdsa_enc'));
 	t.throws(function () {
